flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -493,8 +493,6 @@ private void runMailboxLoop() throws Exception {
 					Thread.currentThread().interrupt();
 					throw interruption.get();
 				}
-			} else if (canceled) {
-				LOG.warn("Error while canceling task.", e);
```

Comment:
nit: condition can now be simplified to `interruption.isPresent() && !canceled`

Reply:
Could it be? There is still need for 3 different branches. Admittedly branches could be re-order to avoid one level of nesting, but still, how could they be squashed?